lasagne/layers/dnn.py

```diff
@@ -220,34 +218,22 @@ def __init__(self, incoming, num_filters, filter_size, stride=(1, 1),
         self.untie_biases = untie_biases
         self.flip_filters = flip_filters
 
-        if border_mode is not None and pad is not None:
-            raise RuntimeError("You cannot specify both 'border_mode' and "
-                               "'pad'. To avoid ambiguity, please specify "
-                               "only one of them.")
-        elif border_mode is None and pad is None:
-            # no option specified, default to valid mode
-            self.pad = (0, 0)
-            self.border_mode = 'valid'
-        elif border_mode is not None:
-            if border_mode == 'valid':
+        if isinstance(pad, basestring):
+            if pad == 'valid':
                 self.pad = (0, 0)
-                self.border_mode = 'valid'
-            elif border_mode == 'full':
-                self.pad = (self.filter_size[0] - 1, self.filter_size[1] - 1)
-                self.border_mode = 'full'
-            elif border_mode == 'same':
+            elif pad == 'full':
+                self.pad = 'full'
```

Comment:
So then maybe my other suggestion for the `if` clauses would be better to avoid the `basestring` check: https://github.com/Lasagne/Lasagne/pull/311#discussion_r33666980